api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -723,8 +724,12 @@ public String getSecretQuestion(User user) throws APIException {
 	 */
 	@Override
 	@Transactional(readOnly = true)
-	public User getUserByEmail(String email) throws APIException {	
-		return dao.getUserByEmail(email);
+	public User getUserByEmail(String email) {
+		if (EmailValidator.getInstance().isValid(email)) {
+			return dao.getUserByEmail(email);
+		}
+		throw new APIException("error.email.invalid", (Object[]) null);
```

Comment:
Please don't pass in null arguments

Reply:
The constructor which interpretes the string passed from the messages expects two arguments. else the error.email.invalid will just be treated as a normal string

Reply:
Please remove it, take a closer look

Reply:
So you're saying this constructor is the only one that translates messages?